front/pages/w/[wId]/members/index.tsx

```diff
@@ -782,19 +805,20 @@ function ChangeMemberModal({
   onClose: () => void;
   member: UserTypeWithWorkspaces | null;
 }) {
+  assert(
```

Comment:
What is this assert? How does this work? 🙏🏻 
And how can user have a none role?

Reply:
It throws an exception if the condition is not met. Here used to make typescript happy -- will add "unreachable" in the comment. User cannot have a none role here because of the filter in the caller, but typescript can't infer that

Reply:
The error message here was indeed unclear

Reply:
And how does that look like on a Front component? Does it display Next raw error message?  

If we know it cannot happen maybe we can simply do a return null if it happens to make typescript happy?

Reply:
Yes, the next error message. The problem with return null is that it's silent. IMHO better to behave as in rest of code, throwing when narrowing type. If something we really think can't happen does indeed happen, let it yell very loud.

Reply:
Did you mean to use `assertNever` that is a pattern on our codebase?

Reply:
To me this is a valid usage: https://github.com/dust-tt/dust/blob/main/front/components/assistant_builder/ActionScreen.tsx#L372-L375

We return empty for none and assertNever to raise only if it's an unknown role. WDYT?

Reply:
The use case is similar to assertNever, but it's not exactly the same. It's a more general situation where we want to tell typescript "this general condition is met and you can use it to narrow type"

The goal is not to raise for unknown role here which type safety correclty prevents (so it does not need additional check). If that was the need, you'd be correct, we should use assertNever

The goal is to raise when role is none, which will tell typescript that in the following code, role cannot be none. 
This is similar to other unreachables we have in our codebase use for type narrowing, e.g.
https://github.com/dust-tt/dust/blob/225405a5c46780be14cf4fcb0f32b59d2df6a25a/front/components/assistant_builder/AssistantBuilder.tsx#L621

https://github.com/dust-tt/dust/blob/858fd81ebe7c63ca0c52b1acffbe104fcfe5814c/front/lib/api/assistant/configuration.ts#L484

https://github.com/dust-tt/dust/blob/c474d6a2c84d18b0711df5a266d5996301696b85/front/lib/api/assistant/messages.ts#L238

We can chat about it IRL tom?

Reply:
No need I don't want to make you loose time, I'm not a fan of this code but you're the PR owner I trust you if you think this is good 👍🏻